### PR TITLE
Set vars in Makefile and bump go to 1.19 to enable FIPS-compliant build

### PR DESCRIPTION
**Pull request by gsleeman**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/observatorium/token-refresher
 
-go 1.14
+go 1.19
```
Comment (saswatamcode) [inline go.mod:3]: I think we need to run `go mod tidy` for this. That'll fix CI! 🙂

File: Makefile [hunk 2]
```diff
@@ -8,6 +8,8 @@ FIRST_GOPATH := $(firstword $(subst :, ,$(shell go env GOPATH)))
 OS ?= $(shell uname -s | tr '[A-Z]' '[a-z]')
 ARCH ?= $(shell uname -m)
 GOARCH ?= $(shell go env GOARCH)
+GOFLAGS := -tags=fips_enabled
```
Comment (saswatamcode) [inline Makefile]: Do we need this build tag here?

File: Makefile [hunk 3]
```diff
@@ -8,6 +8,8 @@ FIRST_GOPATH := $(firstword $(subst :, ,$(shell go env GOPATH)))
...
+GOEXPERIMENT := boringcrypto
```
Comment (saswatamcode) [inline Makefile:11]: Afaiu, this is a cgo impl, with some overhead. Which might be fine for token-refresher use cases.

**Review by saswatamcode — COMMENTED** [review]
Thanks for enabling this! I guess no modifications to TLS flags are needed in this case?

Some comments.

File: Makefile (hunk 2 above)
Comment (gsleeman) [reply Makefile]: Looked a little closer, and I don't think it's necessary

File: Makefile (hunk 3 above)
Comment (gsleeman) [reply Makefile:11]: It doesn't appear to require cgo - the build target specifies CGO_ENABLED=0 and the output is a static binary that includes the boringcrypto symbols

File: go.mod (hunk 1 above)
Comment (gsleeman) [reply go.mod:3]: Done - it seems there's some differences in how go1.19 handles vendor modules, namely it doesn't track the go.{mod,sum} in the module's vendor directory any more.

**Review by saswatamcode — APPROVED** [review]
LGTM!

**Pull request merged**